extensions/amp-carousel/0.1/test/test-slides.js

```diff
@@ -920,7 +920,7 @@ describe('empty Slides functional', () => {
     }).to.throw(/should have at least 1 slide/);
     expect(() => {
       slides.layoutCallback();
-    }).to.not.throw();
+    }).to.throw(/Element expected/);
```

Comment:
that is only to get the extension to fully type check, that assertElement is actually stripped on the production binary. the only place where it would truly error is in buildCallback for the user saying "hey you are using a carousel but have zero slides, that doesn't make sense".

unfortunately a lot of our properties are now nullable when they previously where not because we treated `buildCallback` as our constructor since thats when the custom-element is actually ready and you could query its subtree. to get our code to type check we actually need to have a real `constructor` method now and that is where we need to declare properties and at that point the element isn't ready to be queried yet.